### PR TITLE
[JIT Kernel] Add hadamard kernel test and benchmark

### PR DESCRIPTION
**Pull request by xingsy97**
## Motivation
Add missing test and benchmark coverage for hadamard JIT kernel.
Current `sgl-kernel/tests/test_hadamard.py` only covers the AOT path.

## Modifications
Add `sgl-kernel/tests/test_hadamard_jit.py`.
Add `sgl-kernel/benchmark/bench_hadamard.py`.

## Accuracy Tests
`python -m pytest sgl-kernel/tests/test_hadamard_jit.py -q --tb=short`
passed.

## Benchmarking and Profiling

```
GPU: NVIDIA GeForce RTX 5060 Laptop GPU
Driver: 580.126.20 | CUDA: 12.8 | PyTorch: 2.9.1+cu128
Script: sgl-kernel/benchmark/bench_hadamard.py

hadamard kernel perf (unit: us):
   batch × dim      AOT      JIT      naive (scipy)
   1 × 64           1.30     1.29     2.42
   1 × 1024         1.55     1.79     9.47
   1 × 32768        10.14    10.12    7605.74
   256 × 64         1.63     1.79     3.19
   256 × 1024       5.69     5.77     35.77
   256 × 32768      243.39   244.12   23180.74
```
No regression from AOT to JIT

`naive (scipy)` refers to simple PyTorch result reference built with scipy's hadamard matrix (pad to 2^n, F.linear, truncate), mainly for correctness validation. 

## Checklist
- [x] Format your code according to the [Format code with pre-commit](https://docs.sglang.io/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [x] Add unit tests according to the [Run and add unit tests](https://docs.sglang.io/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [ ] Update documentation according to [Write documentations](https://docs.sglang.io/developer_guide/contribution_guide.html#write-documentations).
- [x] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.io/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.io/developer_guide/contribution_guide.html#benchmark-the-speed).
- [x] Follow the SGLang code style [guidance](https://docs.sglang.io/developer_guide/contribution_guide.html#code-style-guidance).

### CODE REVIEW

File: python/sglang/jit_kernel/tests/test_hadamard_jit.py [hunk 1]
```diff
@@ -0,0 +1,439 @@
...
+---++-+-++-+---+-+++
++---++-+-+--+--++-++
+++---++-+----+-+++-+
+-++---++-+----+++++-
+-+--+--++-+----+----
++-+-----++-+----+---
+-+-+-+---+--+----+--
+--+-+++------+----+-
++--+--++------+----+
+"""
```
[the remark this answers is not in the document]
Comment (xingsy97) [reply python/sglang/jit_kernel/tests/test_hadamard_jit.py:58]: verified it's a square

File: python/sglang/jit_kernel/tests/test_hadamard_jit.py [hunk 2]
```diff
@@ -0,0 +1,439 @@
...
++-+-++++--++-++----+-+-+----++--+--++++-
+++-+-++++--++-++------+-+----++--+--++++
++-+-+-++++--++-++----+-+-+----++--+--+++
++--+-+-++++--++-++---++-+-+----++--+--++
++---+-+-++++--++-++--+++-+-+----++--+--+
++----+-+-++++--++-++-++++-+-+----++--+--
+++----+-+-++++--++-+--++++-+-+----++--+-
++++----+-+-++++--++----++++-+-+----++--+
++-++----+-+-++++--++-+--++++-+-+----++--
+"""
```
[the remark this answers is not in the document]
Comment (xingsy97) [reply python/sglang/jit_kernel/tests/test_hadamard_jit.py:132]: verified it's a square

**Review by BBuf — APPROVED** [review]

File: python/sglang/jit_kernel/benchmark/bench_hadamard.py
Comment (DarkSharpness) [inline python/sglang/jit_kernel/benchmark/bench_hadamard.py:1]: Can we put this benchmark under `python/sglang/jit_kernel/benchmark`?

Comment (xingsy97) [reply python/sglang/jit_kernel/benchmark/bench_hadamard.py:1]: done

File: python/sglang/jit_kernel/benchmark/bench_hadamard.py [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+            naive_fn,
+            quantiles=quantiles,
+        )
+    elif provider == "aot_kernel":
+        ms, min_ms, max_ms = triton.testing.do_bench_cudagraph(
+            lambda: hadamard_transform_aot(x.clone(), scale=scale),
+            quantiles=quantiles,
+        )
+    elif provider == "jit_kernel":
+        ms, min_ms, max_ms = triton.testing.do_bench_cudagraph(
```
Comment (DarkSharpness) [inline python/sglang/jit_kernel/benchmark/bench_hadamard.py]: Could we avoid calling `triton.testing.do_bench_cudagraph` for 3 times. Personally, I would use a mapping to get the differernt function for each provider. You may refer to other JIT kernel benchmark as well.

File: python/sglang/jit_kernel/benchmark/bench_hadamard.py [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+    if provider == "naive":
+        # Precompute Hadamard matrix on GPU to avoid CPU-GPU transfer during
+        # CUDA graph capture.
+        log_dim = math.ceil(math.log2(dim)) if dim > 0 else 0
+        dim_padded = 2**log_dim if dim > 0 else 1
+        H = torch.tensor(
+            hadamard(dim_padded, dtype=float), dtype=dtype, device=device
+        )
+
+        def naive_fn():
```
Comment (DarkSharpness) [inline python/sglang/jit_kernel/benchmark/bench_hadamard.py]: Move this out of `benchmark` function. Maybe rename it to something easier to follow (e.g. `torch_hadamard_transform`.

**Review by DarkSharpness — APPROVED** [review]

File: python/sglang/jit_kernel/benchmark/bench_hadamard.py (hunk 4 above)
Comment (xingsy97) [reply python/sglang/jit_kernel/benchmark/bench_hadamard.py]: updated according to comments, including the previous one.

File: python/sglang/jit_kernel/benchmark/bench_hadamard.py [hunk 5]
```diff
@@ -20,31 +22,32 @@
...
 
-# CI environment detection
-IS_CI = (
-    os.getenv("CI", "false").lower() == "true"
-    or os.getenv("GITHUB_ACTIONS", "false").lower() == "true"
-)
-
-
-# CI environment uses simplified parameters
 if IS_CI:
```
Comment (DarkSharpness) [inline python/sglang/jit_kernel/benchmark/bench_hadamard.py]: maybe align with other benchmarks and use utililiy function in `utils.py`

Comment (xingsy97) [reply python/sglang/jit_kernel/benchmark/bench_hadamard.py]: updated for util func. works well.

**Review by DarkSharpness — APPROVED** [review]
